devtools/server/actors/replay/auth.js

```diff
@@ -73,12 +62,13 @@ function setReplayRefreshToken(token) {
 }
 
 function setReplayUserToken(token) {
-  if (token) {
-    deferredAccessToken.resolve(token);
-  } else {
-    deferredAccessToken = defer();
-  }
-  Services.prefs.setStringPref("devtools.recordreplay.user-token", token || "");
+  token = token || "";
+
+  // Only update and notify if the token has changed
+  if (token === getReplayUserToken()) return;
+
+  Services.prefs.setStringPref("devtools.recordreplay.user-token", token);
+  notifyWebChannelTargets();
```

Comment:
Would we care enough to do this with
```
Services.prefs.addObserver("devtools.recordreplay.user-token", () => {
  notifyWebChannelTargets();
});
```
so it also applies if the user clears the token in `about:config` or something?

Reply:
That seems reasonable to me. :+1: